internal/sbi/processor/processor.go

```diff
@@ -0,0 +1,29 @@
+package processor
+
+import (
+	"github.com/free5gc/udm/internal/sbi/consumer"
+	"github.com/free5gc/udm/pkg/app"
+)
+
+type ProcessorUdm interface {
+	app.App
+
+	Consumer() *consumer.Consumer
+}
+
+type Processor struct {
+	ProcessorUdm
+	consumer *consumer.Consumer
+}
+
+func NewProcessor(udm ProcessorUdm) (*Processor, error) {
```

Comment:
Please refer to https://github.com/free5gc/amf/pull/132/files

You don't need to implement the Consumer getter here.